go/libraries/doltcore/table/editor/index_editor.go

```diff
@@ -287,7 +288,14 @@ func RebuildIndex(ctx context.Context, tbl *doltdb.Table, indexName string, opts
 		return types.EmptyMap, fmt.Errorf("index `%s` does not exist", indexName)
 	}
 
-	rebuiltIndexData, err := rebuildIndexRowData(ctx, tbl.ValueReadWriter(), sch, tableRowData, index, opts)
+	tf := tupleFactories.Get().(*types.TupleFactory)
+
+	tmpDir, err := ioutil.TempDir("", "tempif-*")
```

Comment:
pass down the appropriate tempDir from the sessionState? this fails on linux

Reply:
or use the .dolt/temptf. I'd do whatever we do already for other bulk editors.  I can't remember exactly what we do.